wf-galleon/wildfly_channel_in_zips.adoc

```diff
@@ -0,0 +1,125 @@
+---
+= [Experimental] Update WildFly download distributions using channels
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+[abstract]
+As an user, I can download WildFly from a zip archive and be able to update it to a more recent release.
+
+== Overview
+
+WildFly now publishes https://repo1.maven.org/maven2/org/wildfly/channels/[channels] that is a mechanism to keep a WildFly installation up to date with the latest releases.
+
+This works out of the box if users is provisioning WildFly (using the wildfly-maven-plugin or Prospero).
+However for users that are downloading WildFly from our traditional distributions (zip & tgz archives) they can not take advantage of this mechanism.
+
+If our archives would work out of the box with channels, it would significantely simplify updating an installation.
+
+The workflow would be as simple as:
+
+1. Download WildFly version x.y.z from our download page
+2. Use it as normal
+3. A micro release of WildFly x.y.(z+ 1) is made available
+4. Run a script in the existing WildFly installation to update if from x.y.z to x.y.(z+1)
+5. Reload the server to run your applications on WildFly x.y.(z+ 1)
+
+If there are multiple available updates for WildFly, the user must be able to decide which one they want to update to.
+This lets the user decide whether they want to update to the latest micro release or the next major release.
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-19221
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19130
+
+=== Stability Level
+
+* [X] Experimental
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:{bstansbe@redhat.com}[{Brian Stansberry}]
+* mailto:{bspyrkos@redhat.com}[{Bartosz Spyrko-Smietanko}]
+
+=== Testing By
+
+* [X] Engineering
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped)
+
+== Requirements
+
+* WildFly distributions archives (zip & tgz) must contain the channel metadata that describes their installation state.
+* WildFly distributions archives (zip & tgz) must contain the tooling (as scripts) to let users update their installations
+** Users must be able to list and select the updates to apply to their installations
+** The user MUST specify the updates to apply.
+* Updates must not discard any users changes to an installation (in their configuration files or JBoss modules directory)
+
+=== Non-Requirements
+
+* Trimming an existing installation coming from WildFly distributions with Galleon layers is not supported.
+
+=== Future Work
+
+WIP
+
+== Backwards Compatibility
+
+This enhancement does not affect previous versions of WildFly.
+
+=== Default Configuration
+
+Updating an installation could update its default configuration (eg if the update is to a major version).
```

Comment:
Currently Galleon (and consequently Prospero) will only update default configurations if there are no user changes in that configuration file. Any changes are recorded in a `.new` file and it's up to the user to merged those. 

Is this the expected behaviour or are we going to try to merge the user changes with upcoming changes?

Reply:
Regarding reloading - the changes to the server have to applied while the server is down, so it's a case of starting the server again rather than reloading.

Right now prospero should display a list of such conflicts at the end of update and let the user know they might need to merge those themselves.